Assets/HoloToolkit/SpatialMappingComponent/Support/SMBaseAbstract.cs

```diff
@@ -0,0 +1,600 @@
+        }
+        else
+        {
+            surfaceObserver.SetVolumeAsAxisAlignedBox(transform.position, Extents);
+            bounds.center = transform.position;
+            bounds.extents = Extents;
+        }
+    }
+
+    /// <summary>
+    /// Standard method called when the component is destoryed.
+    /// 
+    /// When this component is destroyed, we clean up all of our tracked GameObjects
+    /// </summary>
+    protected virtual void OnDestroy()
+    {
+        foreach (GameObject go in SpatialMeshObjects.Values)
+        {
+            if (go != null)
+            {
+                FinalDestroy(go);
+            }
+        }
+        SpatialMeshObjects.Clear();
+
+        foreach (RemovedSurfaceHolder rsh in RemovedMeshObjects.Values)
+        {
+            if (rsh.gameObject != null)
+            {
+                FinalDestroy(rsh.gameObject);
+            }
+        }
+        RemovedMeshObjects.Clear();
+    }
+
+    /// <summary>
+    /// Standard method called when the component is disabled
+    /// 
+    /// When this component is disabled, we disable all of the tracked GameObjects
+    /// </summary>
+    protected virtual void OnDisable()
+    {
+        foreach (GameObject go in SpatialMeshObjects.Values)
+        {
+            if (go != null)
+            {
+                go.SetActive(false);
+            }
+        }
+
+        foreach (RemovedSurfaceHolder rsh in RemovedMeshObjects.Values)
+        {
+            if (rsh.gameObject != null)
+            {
+                rsh.gameObject.SetActive(false);
+            }
+        }
+    }
+
+
+    /// <summary>
+    /// Standard method called when the component is enabled
+    /// 
+    /// When this component is enabled, we enable all of the tracked GameObjects
+    /// </summary>
+    protected virtual void OnEnable()
+    {
+        foreach (GameObject go in SpatialMeshObjects.Values)
+        {
+            if (go != null)
+            {
+                go.SetActive(true);
+            }
+        }
+
+        foreach (RemovedSurfaceHolder rsh in RemovedMeshObjects.Values)
+        {
+            if (rsh.gameObject != null)
+            {
+                rsh.gameObject.SetActive(true);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Handler for when the SurfaceObserver completes RequestMeshAsync
+    /// 
+    /// The base class defines this function to make it easier for subclasses to modify meshes upon completion without needing to handle the actual processing
+    /// </summary>
+    /// <param name="bakedData">The processed SurfaceData</param>
+    /// <param name="outputWritten">Whether or not output was written</param>
+    /// <param name="elapsedBakeTimeSeconds">The time in seconds it took to request and populate the mesh</param>
+    protected virtual void SurfaceObserver_OnDataReady(SurfaceData bakedData, bool outputWritten, float elapsedBakeTimeSeconds)
+    {
```

Comment:
I'm sorry, I'm not familiar with that concept. Could you explain more/link to an explanation/post example code?
Thanks